text/0521-find-by-identifier.md

```diff
@@ -0,0 +1,122 @@
+- Start Date: (fill me in with today's date, 2019-08-29)
+- Relevant Team(s): EmberData
+- RFC PR: https://github.com/emberjs/rfcs/pull/521
+- Tracking: (leave this empty)
+
+# [DATA] findRecord/peekRecord via Identifier
+
+## Summary
+
+Users should be able to peek or find records based on `lid`.
+
+## Motivation
+
+Apps and Addons making use of [Identifiers](https://github.com/emberjs/rfcs/pull/403)
+may wish to peek or find the record for a known `identifier`. This
+RFC would allow them to do so.
```

Comment:
Minor quibble here, but since the existing APIs are deprecated in this RFC (AFAICT) this doesn't _allow_ folks to use identifiers it _requires_ it...

Reply:
You are misreading the signature

```
ExistingResourceObject | NewResourceObject | Identifier
```

This allows for three distinct scenarios:

- `ExistingResourceObject` would allow for "just type + id" which is the current signature,
- `NewResourceObject` allows you to be able to find again a record you know you have newly created but perhaps not held on to a reference for. It does so by utilizing `lid`.
- `Identifier` allows you to lookup by "just lid" without type or id being required.

Reply:
we decided to not deprecate the old signatures for now as well